src/pressure/VCycleMG.cpp

```diff
@@ -205,45 +131,31 @@ VCycleMG::~VCycleMG() {
 /// \param  b       rhs
 /// \param  sync    synchronization boolean (true=sync (default), false=async)
 // ************************************************************************
-void VCycleMG::UpdateInput(Field *out, Field *b, bool sync) {
-    auto domain = Domain::getInstance();
+void VCycleMG::UpdateInput(Field &out, Field const &b, bool sync) {
     // local variables and parameters for GPU
-    auto d_out = out->data;
-    auto s_out = domain->get_size(out->get_level());
-    auto d_b = b->data;
-    auto s_b = domain->get_size(b->get_level());
-
-    auto f_err1 = error1[0];
-    auto f_mg_tmp = mg_temporal_solution[0];
-    auto f_res1 = residuum1[0];
-
-    auto d_err1 = error1[0]->data;
-    auto d_mg_tmp = mg_temporal_solution[0]->data;
-    auto d_res1 = residuum1[0]->data;
-
-    auto s_err1 = domain->get_size(error1[0]->get_level());
-    auto s_mg_tmp = domain->get_size(mg_temporal_solution[0]->get_level());
-    auto s_res1 = domain->get_size(residuum1[0]->get_level());
+    auto err1 = error1[0];
+    auto mg_tmp = mg_temporal_solution[0];
+    auto res1 = residuum1[0];
 
     auto boundary = BoundaryController::getInstance();
     size_t *d_iList = boundary->get_innerList_level_joined();
     auto bsize_i = boundary->getSize_innerList();
 
     // use iList on level 0, since update on level 0
-#pragma acc kernels present(d_out[:s_out], d_b[:s_b], d_err1[:s_err1], \
-                            d_mg_tmp[:s_mg_tmp], d_res1[:s_res1], d_iList[:bsize_i]) async
+#pragma acc kernels present(out, b, err1, \
+                            mg_tmp, res1, d_iList[:bsize_i]) async
     {
 #pragma acc loop independent
         for (size_t j = 0; j < bsize_i; ++j) {
             const size_t i = d_iList[j];
-            d_err1[i] = d_out[i];
-            d_mg_tmp[i] = d_out[i];
-            d_res1[i] = d_b[i];
+            (*err1)[i] = out[i];
+            (*mg_tmp)[i] = out[i];
+            (*res1)[i] = b[i];
         }
         if (sync) {
```

Comment:
probably the copy function?

Reply:
i would like to, but unfortunately those are pointers to arrays and not fields.

Reply:
we could open an investigation issue to investigate whether it is possible to use fields and the copy function here. Probably at some point, a field was constructed and the data extracted

Reply:
with regard to the definition of `err1`, `mg_tmp` and `res1` those are vectors with Field objects, therefore `(*err1)[i]` should be eligible for the copy function of the Field class. Or am I overlooking something?

Reply:
I have to correct myself the problem is, we are coping only the innerlists.

Reply:
will be changed with PR #70